Chore: Standardize Datadog Environment Variables

ahosking opened the pull request:
Mesh and JBX have always been 'prod' in Datadog where everything else is 'production'.
This has created some consistently annoying observability and visibility blockers.

This change allows us to leave our existing environment variables intact through opsworks declarations and cookbooks while improving our use inside of Datadog.

Bellardia commented on .chef/client_amazon.rb:
```diff
@@ -0,0 +1,27 @@
+current_dir                     = File.dirname(__FILE__)
```
merge this with client.rb

ahosking replied:
This has evaporated because it shouldn't have been paired with the Datadog Changes, fyi.

Bellardia commented on configure/recipes/datadog.rb:
```diff
@@ -1,4 +1,4 @@
-node.default["datadog"]["tags"]["environment"] = node["environment"]
+node.default["datadog"]["tags"]["environment"] = (node["environment"].include? "prod" ? "production": node["environment"])
```
can you introduce this change independently from this PR which is otherwise entirely around systems manager

ahosking replied:
Ahhh, fiddlesticks!
Absolutely, I don't know how I missed that.
My apologies!

Commit:
Chore: Standardize Datadog Environment Variables

Mesh and JBX have always been 'prod' in Datadog where everything else is 'production'.
This has created some consistently annoying observability and visibility blockers.

This change allows us to leave our existing environment variables intact through opsworks declarations and cookbooks while improving our use inside of Datadog.
